lib/dns.js

```diff
@@ -82,11 +82,36 @@ function onlookup(err, addresses) {
 }
 
 
+function onlookupall(err, addresses) {
+  var i, results = [];
+  if (err) {
+    return this.callback(errnoException(err, 'getaddrinfo', this.hostname));
+  }
+
+  for (i = 0; i < addresses.length; i++) {
+    if (this.family) {
```

Comment:
Looks like your suggestion returns `family = 6` on `family = 4` queries. It'm still investigating why, but a ternary seems to work:
````
family: this.family ? this.family : addresses[i].indexOf(':') >= 0 ? 6 : 4
```